docs: clarify readiness probe for replicas (#6629)
cloudnative-pg/cloudnative-pg: docs/src/instance_manager.md

of the Pod, the instance manager acts as a backend to handle the
[startup, liveness and readiness probes](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes).

## Startup, Liveness, and Readiness Probes

CloudNativePG leverages [PostgreSQL's `pg_isready`](https://www.postgresql.org/docs/current/app-pg-isready.html)
to implement Kubernetes startup, liveness, and readiness probes.

### Startup Probe

The startup probe ensures that a PostgreSQL instance, whether a primary or
standby, has fully started according to `pg_isready`.
While the startup probe is running, the liveness and readiness probes remain
disabled. Following Kubernetes standards, if the startup probe fails, the
kubelet will terminate the container, which will then be restarted.

> The startup probe provided by CloudNativePG relies on `.spec.startDelay`.
> The `.spec.startDelay` parameter specifies the maximum time, in seconds,

Reviewer: I don't like "relies on" here.
How about something like this:
```suggestion
The startup probe provided by CloudNativePG is configurable via the
parameter `.spec.startDelay`, which specifies the maximum time, in seconds,
```

### Liveness Probe

The liveness probe starts once the startup probe has successfully completed.

> Its purpose is to verify that the PostgreSQL instance, whether primary or
> standby, is functioning correctly by relying on the `pg_isready` check.

Reviewer: ```suggestion
Its purpose is to verify that the PostgreSQL instance, whether primary or
standby, is functioning correctly according to the `pg_isready` check.
```

### Readiness Probe

The readiness probe begins once the startup probe has successfully completed.
Its purpose is to check whether the PostgreSQL instance is ready to accept
traffic and serve requests.

> For streaming replicas, it also requires the WAL receiver to start successfully.

Reviewer: I find the explanation clearer in the `replication.md` file warning box:
> when it has successfully connected to the source at least once.
```suggestion
For streaming replicas, it also requires that they have connected to the source at least once.
```
